### PR TITLE
Electricity Maps (Independent Publisher)

### PR DESCRIPTION
**Pull request by VitaliiSorokin**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 

test_section

paconn_valid
flow_1
flow_2
flow_3

### CODE REVIEW

**Comment by VitaliiSorokin** [comment]
@microsoft-github-policy-service agree [company="{your company}"]

**Comment by VitaliiSorokin** [comment]
@microsoft-github-policy-service agree

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,1239 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "description": "You can use Electricity Maps API to get access to information about where the electricity in a specific area comes from, how much carbon was emitted to produce it, and to get historical data, live data and forecasts.",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:4]: Hi @VitaliiSorokin, Please remove the word API from the description. please see our docs [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#write-a-description-for-your-connector)

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,1239 @@
...
+    "title": "Electricity Maps",
+    "version": "0.1",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:6]: @VitaliiSorokin please change the version to "1.0"

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,1239 @@
...
+  "paths": {
+    "/carbon-intensity/forecast": {
+      "get": {
+        "parameters": [
+          {
+            "name": "zone",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "A string representing the zone identifier",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:27]: @VitaliiSorokin all parameters' descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#parameters)
Please fix this in all parameters in the swagger file

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,1239 @@
...
+                  "x-ms-summary": "Zone"
+                }
+              },
+              "type": "object"
+            }
+          }
+        },
+        "operationId": "BreakdownLatest",
+        "summary": "Get Live Power Breakdown",
+        "description": "Retrieves the last known data about the origin of electricity in an area"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:859]: @VitaliiSorokin all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 5]
```diff
@@ -0,0 +1,1239 @@
...
+                  "x-ms-summary": "Zone"
+                }
+              },
+              "type": "object"
+            }
+          }
+        },
+        "operationId": "ConsumptionForecast",
+        "summary": "Get Power Consumption Forecast",
+        "description": "Retrieves the forecasted power consumption breakdown of an area, which represents the physical origin of electricity broken down by production type. It can either be queried by zone identifier or by geolocation"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:988]: @VitaliiSorokin all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,1239 @@
...
+                  "x-ms-summary": "Zone"
+                }
+              },
+              "type": "object"
+            }
+          }
+        },
+        "operationId": "BreakdownForecast",
+        "summary": "Get Power Production Forecast",
+        "description": "Retrieves the forecasted power production breakdown of an area by production type. It can either be queried by zone identifier or by geolocation"
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:1122]: @VitaliiSorokin all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 7]
```diff
@@ -0,0 +1,1239 @@
...
+                  },
+                  "description": "Zone List",
+                  "x-ms-summary": "Zones"
+                }
+              }
+            }
+          }
+        },
+        "summary": "Get Available Zones",
+        "description": "A list of countries and zones by code (zone being the country is divided by zones seen on app.electricitymap.org)",
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:1169]: @VitaliiSorokin all operations descriptions should end with a period, No period required only if there's a single word. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/connector-string-guidance#operations)

File: independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json [hunk 8]
```diff
@@ -0,0 +1,1239 @@
...
+    {
+      "propertyName": "Privacy policy",
+      "propertyValue": "https://www.electricitymaps.com/privacy-policy"
+    },
+    {
+      "propertyName": "Categories",
+      "propertyValue": "Data"
+    }
+  ],
+  "x-components": {}
```
Comment (Amjed-Ayoub) [inline independent-publisher-connectors/Electricity Maps/apiDefinition.swagger.json:1238]: @VitaliiSorokin please remove this line unless you have a reason to keep it.

**Review by Amjed-Ayoub — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
[[certify-connector]]

**Comment by Amjed-Ayoub** [comment]
Hello @VitaliiSorokin,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by VitaliiSorokin** [comment]
Hello @Amjed-Ayoub,
Thank you for the update.
I just want to clarify one moment: at what point and by whom my pull request should be merged to the dev branch?

**Comment by Amjed-Ayoub** [comment]
Hi @VitaliiSorokin, After the connector is deployed we will merge the pull request. you can track the progress using this conversation. we will keep you posted. 
You should see a label called "deployment-in-progress" once the deployment starts. 
Please let me know if you have any questions.
Thank you so much again.

**Comment by Amjed-Ayoub** [comment]
Hello @VitaliiSorokin,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

**Comment by VitaliiSorokin** [comment]
Great news!
Thank you @Amjed-Ayoub!
It was a pleasure to work with you. Looking forward to future cooperation.

**Comment by Amjed-Ayoub** [comment]
Hello @VitaliiSorokin,
I hope you are doing well.
Congratulations, the deployment of your connector and documentation page was already completed, We have a short [survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbR0BHMrjhL0hDmckROosW6AFURERIVUUzMk5XRzJGV1dJM1Q5RVZJUjFXSC4u) on the connector certification process that we would like for you to complete. Your feedback is appreciated and will help us improve the program.
Thank you very much for working with us.

**Review by Amjed-Ayoub — APPROVED** [review]

**Pull request merged**